bundles/org.openhab.ui/web/src/assets/definitions/transformations.js

```diff
@@ -1,14 +1,25 @@
 export default {
   SNIPPETS: {
-    SCRIPT: {
-      'application/vnd.openhab.dsl.rule': 'var returnValue = "String has " + input.length + " characters"\n\nreturnValue',
-      'application/javascript': '(function(data) {\n  var returnValue = "String has " + data.length + " characters"\n  return returnValue\n})(input)',
-      'application/javascript;version=ECMAScript-5.1': '(function(data) {\n  var returnValue = "String has " + data.length + " characters"\n  return returnValue\n})(input)',
-      'application/x-ruby': '"String has #{input.length} characters"'
-    }
+    DSL: 'var returnValue = "String has " + input.length + " characters"\n\nreturnValue',
+    JINJA: '{# Available values:\nvalue - The incoming value.\nvalue_json - The incoming value parsed as JSON. #}\n{{value_json[\'AM2301\'].Temperature}}',
+    JS: '(function(data) {\n  var returnValue = "String has " + data.length + " characters"\n  return returnValue\n})(input)',
+    NASHORNJS: '(function(data) {\n  var returnValue = "String has " + data.length + " characters"\n  return returnValue\n})(input)',
+    MAP: 'key=value\n=default',
+    RB: '"String has #{input.length} characters"',
+    XSLT: '<?xml version=\'1.0\' encoding=\'UTF-8\'?>\n' +
+      '<xsl:stylesheet version=\'2.0\' xmlns:xsl=\'http://www.w3.org/1999/XSL/Transform\'>\n' +
+      '   <xsl:output method=\'xml\' indent=\'no\'/>\n' +
+      '   <xsl:template match=\'/\'>\n' +
+      '      <reRoot><reNode><xsl:value-of select=\'/root/node/@val\' /> world</reNode></reRoot>\n' +
+      '   </xsl:template>\n' +
+      '</xsl:stylesheet>'
   },
   EDITOR_MODES: {
+    DSL: 'application/vnd.openhab.dsl.rule',
```

Comment:
Do we need `JS` here? Otherwise LGTM.

Reply:
No, we don't.

When a transformation is loaded, the UI checks if there is an editor mode defined in this file. If it is there this is passed to `script-editor.vue`, if it is not there, the transformation type is used.
The UI's code editor (codemirror) supports various language names out of the box, e.g. xml, and `script-editor.vue` translates language shortcuts such as js, py, rb to support them out of the box.

The `EDITOR_MODES` definitions are only required for transformation services where the name does not securely indicate the MIME type or language name-